services/property-svc/internal/property/aggregate/aggregate.go

```diff
@@ -182,15 +184,56 @@ func (a *PropertyAggregate) onFieldTypeDataSelectOptionsUpserted(evt hwes.Event)
 		return err
 	}
 
-	if a.Property.FieldTypeData.SelectData != nil {
-		// TODO: Check and update already existing SelectOptions don't just append them
-		a.Property.FieldTypeData.SelectData.SelectOptions = append(a.Property.FieldTypeData.SelectData.SelectOptions, payload.UpsertedSelectOptions...)
-	} else {
-		a.Property.FieldTypeData.SelectData = &models.SelectData{
-			SelectOptions: payload.UpsertedSelectOptions,
+	var initialAllowFreetext bool
+	existingSelectOptions := make(map[uuid.UUID]models.SelectOption)
```

Comment:
We update this value later on, at which point the name `existingSelectOptions` is not appropriate anymore